packages/models-library/src/models_library/settings/base.py

```diff
@@ -0,0 +1,35 @@
+from contextlib import suppress
+from typing import List, Tuple
+
+from pydantic import BaseSettings, Extra, SecretStr, ValidationError
+
+
+class BaseCustomSettings(BaseSettings):
+    class Config:
+        case_sensitive = False
+        extra = Extra.forbid
+        allow_mutation = False
+        frozen = True
+        validate_all = True
+        json_encoders = {SecretStr: lambda v: v.get_secret_value()}
+
+    @classmethod
+    def set_defaults_with_default_constructors(
+        cls, default_fields: List[Tuple[str, "BaseCustomSettings"]]
+    ):
```

Comment:
I see that you manually build the settings.
What if someone adds a new module and forgets to bind it to this function.

Could we perhaps have an auto discovery for settings classes? At that point we could use the auto discovery in a test to point to the user he needs to plug the value into this method.

Reply:
yes, every settings should be a derived from ``BaseCustomSettings`` but mostly for the pydantic ``Config``. This function is only for the app ``Settings`` to be used. 

It is a good idea to check that all sub-settings are actually derived from the right class. Nonetheless it will have to wait until i have all the settings int he ``models_library.settings`` refactored first. but i take note!